src/Physics/InitialField.cpp

```diff
@@ -1,148 +1,70 @@
 #include <cmath>
 #include <array>
+#include <numeric>
+#include <Eigen/Eigenvalues>
 
 #include <Kernels/precision.hpp>
 #include <Physics/InitialField.h>
+#include <Model/Setup.h>
+#include <Model/common.hpp>
 #include <yateto/TensorView.h>
+#include <utils/logger.h>
+#include <Solver/Interoperability.h>
 
-seissol::physics::Planarwave::Planarwave(double phase)
-  : m_setVar(2),
-    m_varField{1,8},
+extern seissol::Interoperability e_interoperability;
+
+seissol::physics::Planarwave::Planarwave(seissol::model::Material material, double phase)
+  : m_varField{1,8},
     m_ampField{1.0, 1.0},
-    m_kVec{6.283185307179587E-002, 6.283185307179587E-002, 6.283185307179587E-002},
-    m_phase(phase)
+    m_phase(phase),
+    m_kVec({M_PI, M_PI, M_PI})
 {
-  const auto rho0 = 1.0;
-  const auto mu = 1.0;
-  const auto lambda = 2.0;
-
-  // Eigenvalues.
-  const auto cp = std::sqrt((lambda+2*mu)/(rho0));
-  const auto cs = std::sqrt((mu)/(rho0));
-
-  double kVecNorm = 0.0;
-  for (auto k: m_kVec) {
-    kVecNorm += k * k;
+  assert(m_varField.size() == m_ampField.size());
+
+  std::complex<double> planeWaveOperator[NUMBER_OF_QUANTITIES*NUMBER_OF_QUANTITIES];
+  seissol::model::getPlaneWaveOperator(material, m_kVec.data(), planeWaveOperator);
+
+  using Matrix = Eigen::Matrix<std::complex<double>, NUMBER_OF_QUANTITIES, NUMBER_OF_QUANTITIES, Eigen::ColMajor>;
+  Matrix op(planeWaveOperator);
+  Eigen::ComplexEigenSolver<Matrix> ces;
+  ces.compute(op);
+  
+  auto eigenvalues = ces.eigenvalues();
+  std::vector<size_t> sortedIndices(NUMBER_OF_QUANTITIES);
+  std::iota(sortedIndices.begin(), sortedIndices.end(), 0);
+  std::sort(sortedIndices.begin(), sortedIndices.end(), [&eigenvalues](size_t a, size_t b) {
+    return eigenvalues[a].real() < eigenvalues[b].real();
+  });
+
```

Comment:
Could you maybe add a comment that explains that sortedIndices contains a sorted list of eigenvalues-indices. Maybe also why you're sorting them?

Reply:
There is even a comment in the code: sort eigenvalues so that we know which eigenvalue corresponds to which mode
:-D

Reply:
I've added that after Lukas's comment ;-)